be/src/exec/stream/state/imt_state_table.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "exec/stream/state/imt_state_table.h"
+
+namespace starrocks::stream {
+
+IMTStateTable::IMTStateTable(const TIMTDescriptor& imt) : _imt(imt) {
+    DCHECK_EQ(_imt.imt_type, TIMTType::OLAP_TABLE);
+    DCHECK(_imt.__isset.olap_table);
+
+    _db_name = _imt.olap_table.db_name;
+    _table_name = _imt.olap_table.table_name;
+}
+
+Status IMTStateTable::prepare(RuntimeState* state) {
+    // prepare read
+    _table_reader = std::make_unique<TableReader>(_convert_to_reader_params());
+    _schema = _table_reader->tablet_schema();
+    auto field_names = _schema.field_names();
+    _non_keys_field_names = std::vector<std::string>(field_names.begin() + _schema.num_key_fields(), field_names.end());
+
+    // prepare write
+    Status status;
+    _olap_table_sink = std::make_unique<stream_load::OlapTableSink>(&_obj_pool, _output_exprs, &status, state);
+    RETURN_IF_ERROR(status);
+    _olap_table_sink->init(_convert_to_sink_params(), state);
+    return _olap_table_sink->prepare(state);
+}
+
+Status IMTStateTable::open(RuntimeState* state) {
+    return _olap_table_sink->open(state);
+}
+
+TableReaderParams IMTStateTable::_convert_to_reader_params() {
+    return TableReaderParams{
+            .schema = _imt.olap_table.schema,
+            .version = _imt.olap_table.schema.version,
```

Comment:
After new data is ingested into IMT in the last epoch, maybe we should update this version to the newest in the next epoch so that we can read the newest data.

Reply:
So, the `StateTable` and `IMTStateTable` need to be aware of `Epoch`, including `being_epoch` and `commit_epoch`, like the `StreamOperators` ?

Reply:
I doubt only `version` is required for `TableReader`.

Reply:
Ok. Just leave it and extend it in the future if necessary.